components/patient-data/migrations/src/main/java/org/phenotips/data/internal/R71490PhenoTips1280DataMigration.java

```diff
@@ -226,10 +262,24 @@ private void migrateGenes(XWikiDocument doc, DocumentReference oldGenesClassRefe
             } else if (geneComments != null) {
                 String commentAppend = "Automatic migration: \ngene was duplicated in the " + status + " gene section.";
                 commentAppend += "\nOriginal comment: \n" + geneComments;
-                updateComment(geneName, doc, commentAppend, geneClassReference);
+                updateComment(geneName, doc, commentAppend, this.geneClassReference);
             }
         }
-        doc.removeXObjects(oldGenesClassReference);
+        doc.removeXObjects(docRef);
+
+        // fix for PT-3033: users occasionally used the "rejected genes" section in 1.2 only for comments
+        if (StringUtils.isNotBlank(freeComments)) {
+            this.commentsFromEmptyGenes +=
```

Comment:
This isn't reset to the empty string at the start of a patient, so comments will propagate from a patient to the following.